server.py

```diff
@@ -113,5 +129,17 @@ def get_verified_email(jwt):
     return {'email': payload['sub']}
 
 
-host = 'localhost' if 'localhost' in META['RP_ORIGIN'] else '0.0.0.0'
-run(server='gunicorn', host=host, port=int(getenv('PORT', '8080')))
+HANDLERS = {'index': index, 'login': login, 'static': static}
+STATUS = {200: '200 OK', 400: '400 Bad Request'}
+
+def application(env, respond):
+    pi = [] if not env['PATH_INFO'] else env['PATH_INFO'].strip('/').split('/')
+    handler = pi[0] if pi and pi[0] else 'index'
+    status, headers, content = HANDLERS[handler](env)
+    respond(STATUS[status], list(headers.items()))
+    return [content]
+
+
+if __name__ == '__main__':
+    host, port = parse.urlparse(META['RP_ORIGIN']).netloc.split(':')
+    simple_server.make_server(host, int(port), application).serve_forever()
```

Comment:
Scenario: We hit the top of Hacker News, and get a few dozen simultaneous users trying out the demo. Which breaks. Cue sad trombone.

>  As a demo RP, having the lowest possible barrier to access is better

That's a great point. I've been prioritizing legibility in the Demo RP over re-usability, under the presumption that people are more likely to want to skim the code to see what it looks like than they are to actually want to re-use that code directly. Should we prioritize re-use instead?